api/logic/evm.go

```diff
@@ -0,0 +1,40 @@
+package logic
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	ethCommon "github.com/ethereum/go-ethereum/common"
+)
+
+func EVMParseData(data []byte, contractABI *abi.ABI) (*abi.Method, []interface{}, error) {
```

Comment:
It's not clear what Go types will be used in the elements of `[]interface{}`.

Looking at the ethereum-go tests, it's "whatever makes sense" https://github.com/ethereum/go-ethereum/blob/9231770811cda0473a7fa4e2bccc95bf62aae634/accounts/abi/abi_test.go#L810-L813 but it's not clearly documented. Going more into the weeds, it is one of the many possible return values of [toGoType()](https://github.com/ethereum/go-ethereum/blob/ea9e62ca3db5c33aa7438ebf39c189afd53c6bf8/accounts/abi/unpack.go#L222).

Perhaps we don't need to care much, as long as serializes cleanly into JSON (which is the format we use in the DB, and in our API).
Please test at least with large integers; ethereum-go uses `*big.Int`, which is reasonable, but does not support pgx-compatible serialization. We'll be converting everything into JSON manually before throwing it at pgx, so we're probably fine?
Anyway, it would be good to have some EVM parsing tests not just into Go-native types, but into JSON. If you can find/generate example inputs for wacky types (e.g. multi-dimensional array of tuples of (bigint, address)), all the better.

Reply:
yeah it's *big.Int and it tries to put it into json as just the digits :scream:

Reply:
alright tests are added